pkg/converter/tool/feature.go

```diff
@@ -8,106 +8,144 @@ package tool
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/exec"
 	"regexp"
+	"strings"
 	"sync"
 
 	"github.com/sirupsen/logrus"
-	"golang.org/x/mod/semver"
 )
 
-const envNydusDisableTar2Rafs = "NYDUS_DISABLE_TAR2RAFS"
+type Feature string
+type Features map[Feature]struct{}
 
-var currentVersion string
-var currentVersionDetectOnce sync.Once
-var disableTar2Rafs = os.Getenv(envNydusDisableTar2Rafs) != ""
+const envNydusDisableTar2Rafs string = "NYDUS_DISABLE_TAR2RAFS"
 
 const (
 	// The option `--type tar-rafs` enables converting OCI tar blob
 	// stream into nydus blob directly, the tar2rafs eliminates the
 	// need to decompress it to a local directory first, thus greatly
 	// accelerating the pack process.
 	FeatureTar2Rafs Feature = "--type tar-rafs"
+	// The option `--batch-size` enables merging multiple small chunks
+	// into a big batch chunk, which can reduce the the size of the image
+	// and accelerate the runtime file loading.
+	FeatureBatchSize Feature = "--batch-size"
 )
 
-var featureMap = map[Feature]string{
-	FeatureTar2Rafs: "v2.2",
+var requiredFeatures Features
+var detectedFeatures Features
+var detectFeaturesOnce sync.Once
+var disableTar2Rafs = os.Getenv(envNydusDisableTar2Rafs) != ""
+
+// IsLegal checks the compatibility of Feature with regular expressions.
+// Feature should only contain letters, numbers, hyphens, underlines, and spaces.
+func (feature Feature) IsLegal() bool {
+	re := regexp.MustCompile(`[\w- ]+`)
```

Comment:
Maybe move this out?

Reply:
`IsLegal()` is removed. Because there's no dynamic regular expression compilation anymore.